File: youlexuan-shop-web/src/main/webapp/js/controller/sellerController.js
app.controller('sellerController',function ($scope,sellerService) {

    /*//继承baseController
    $controller('baseController',{$scope:$scope});*/

    //注册对象
    $scope.entity={};

    //密码对象
    $scope.rawPassword="";
    $scope.newPassword="";
    $scope.againPassword="";

    //注册
    $scope.register=function () {

        sellerService.register($scope.entity).success(function (result) {

            if(result.success){
                alert(result.message);
                location.href="shoplogin.html";
                //清一下
                $scope.entity={};
            }else{
                alert(result.message);
            }
        })
    };
    //验证密码
    $scope.checkRawPassword=function () {

        //验证原密码是否正确
        sellerService.checkRawPassword($scope.rawPassword).success(function (bol) {

            if (bol == 'true') {
                //输入原密码正确

                if($scope.newPassword != "" && $scope.againPassword != ""){
                    if ($scope.newPassword == $scope.againPassword) {

                        sellerService.updateNewPassword($scope.newPassword).success(function (bool) {
                            if (bool == 'true') {
                                alert("修改密码成功!");
                                //注销登录
                                parent.location.href = "/logout";
                            } else {
                                alert("修改密码失败!");
                            }
                        })
                    }else{
                        //新密码与确认密码输入不一致
                        alert("新密码与确认密码输入不一致!");
                    }
                }else{
                    alert("新密码和确认密码不能为空!");
                }

            } else {
                alert("原密码输入有误,请重新输入!");
            }

        })
    };
    //返回index页面
    $scope.goIndex=function () {

        parent.location.href="index.html";
    };

    //修改seller资料
    $scope.updateSeller=function () {
        sellerService.updateSeller($scope.entity).success(function (result) {

            if(result.success){
                alert(result.message);
                parent.location.href="index.html";
            }else{
                alert(result.message);
            }
        })
    };
    //修改回显
    $scope.getSeller=function () {

        sellerService.getSeller().success(function (seller) {

            $scope.entity=seller;
            alert($scope.entity.nickName);
        })
    }
});

File: youlexuan-shop-web/src/main/java/com/offcn/shop/controller/SellerController.java
package com.offcn.shop.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.pojo.TbSeller;
import com.offcn.seller.service.SellerService;
import com.offcn.vo.resp.CommonResult;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/seller")
public class SellerController {

    @Reference
    private SellerService sellerService;

    @RequestMapping("/register")
    public CommonResult register(@RequestBody TbSeller tbSeller){

        try{
            //密码加密
            BCryptPasswordEncoder encoder=new BCryptPasswordEncoder();
            if(tbSeller!=null && tbSeller.getPassword()!=null && !tbSeller.getPassword().equals("")){
                String rawPassword=tbSeller.getPassword();
                String password=encoder.encode(rawPassword);
                tbSeller.setPassword(password);
            }
            sellerService.register(tbSeller);
            return new CommonResult(true,"注册成功",tbSeller);
        }catch (Exception e){
            e.printStackTrace();
            return new CommonResult(false,"注册失败",tbSeller);
        }
    }

    @RequestMapping("/checkRawPassword")
    public boolean checkRawPassword(String rawPassword){

        //从security上下文对象中获取name
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        TbSeller tbSeller=sellerService.getSellerBySellerId(name);

        BCryptPasswordEncoder encoder=new BCryptPasswordEncoder();
        boolean bol = encoder.matches(rawPassword, tbSeller.getPassword());
        return bol;
    }

    @RequestMapping("/updateNewPassword")
    public boolean updateNewPassword(String newPassword){

        BCryptPasswordEncoder encoder=new BCryptPasswordEncoder();
        String password = encoder.encode(newPassword);

        //从security上下文对象中获取name
        String name = SecurityContextHolder.getContext().getAuthentication().getName();

        //获取sellerId
        return sellerService.updateNewPassword(password,name);
    }

    @RequestMapping("/updateSeller")
    public CommonResult updateSeller(@RequestBody TbSeller seller){

        try {
            //从上下文获取sellerId
            String name = SecurityContextHolder.getContext().getAuthentication().getName();
            seller.setSellerId(name);
            sellerService.updateSeller(seller);
            return new CommonResult(true,"保存成功",null);
        } catch (Exception e) {
            e.printStackTrace();
            return new CommonResult(false,"保存失败",null);
        }
    }

    @RequestMapping("/getSeller")
    public TbSeller getSeller(){

        //从上下文获取sellerId
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        return sellerService.getSellerBySellerId(name);
    }
}
